### PR TITLE
Ban classification separated into two stages

### PR DESCRIPTION
**Pull request by Michael-Phoenix**
First rawReason gets checked and if classification failed there, the ban.rawNotes get scraped as well.

### CODE REVIEW

File: ban-importer/src/ban-fetcher.js [hunk 1]
```diff
@@ -122,8 +122,9 @@ export default class BanFetcher {
             expires: expires,
             expired: !(ban.attributes.expires === null || expires.getTime() > Date.now()),
 
-            reason: classifyBanReason(`${ban.attributes.reason} ${ban.attributes.note}`),
-            rawReason: `${ban.attributes.reason} ${ban.attributes.note}`,
+            reason: classifyBanReason(`${ban.attributes.reason}`,`${ban.attributes.note}`),
```
Comment (Thomas-Smyth) [inline ban-importer/src/ban-fetcher.js:125]: No need for the empty templating string here, i.e.:
```js
reason: classifyBanReason(ban.attributes.reason, ban.attributes.note),
```

File: ban-importer/src/ban-fetcher.js [hunk 2]
```diff
@@ -122,8 +122,9 @@ export default class BanFetcher {
...
+            rawReason: `${ban.attributes.reason}`,
+            rawNotes: `${ban.attributes.note}`,
```
Comment (Thomas-Smyth) [inline ban-importer/src/ban-fetcher.js:127]: As above.

File: scripts/update-unknown.js [hunk 3]
```diff
@@ -8,11 +8,11 @@ async function main() {
   console.log(`${bans.length} Unknown Bans To Classify`);
   for (const ban of bans) {
     const old = ban.reason;
-    ban.reason = classifyBanReason(ban.rawReason);
     if (old !== ban.reason) {
       console.log(old, ban.reason);
       console.log(`Updating Ban: ${ban.reason}`);
       await ban.save();
+    ban.reason = classifyBanReason(ban.rawReason, ban.rawNote);
```
Comment (Thomas-Smyth) [inline scripts/update-unknown.js]: Was this change intentional? It doesn't seem right, as with this change ban reasons will never be updated if they are updated on BM.

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]

File: scripts/update-unknown.js (hunk 3 above)
Comment (Michael-Phoenix) [reply scripts/update-unknown.js]: fixed in https://github.com/CommunityBanList/Communitybanlist/pull/11/commits/239588b9eac41e7fbb045414fe98a902f321753c

File: ban-importer/src/ban-fetcher.js (hunk 2 above)
Comment (Michael-Phoenix) [reply ban-importer/src/ban-fetcher.js]: fixed in https://github.com/CommunityBanList/Communitybanlist/pull/11/commits/40c517447dfe396de588d13c5dd41cd70efc04b9

File: ban-importer/src/ban-fetcher.js (hunk 1 above)
Comment (Michael-Phoenix) [reply ban-importer/src/ban-fetcher.js]: fixed in https://github.com/CommunityBanList/Communitybanlist/pull/11/commits/40c517447dfe396de588d13c5dd41cd70efc04b9

**Review by Thomas-Smyth — APPROVED** [review]
LGTM.

File: ban-importer/src/ban-fetcher.js [hunk 4]
```diff
@@ -122,8 +122,9 @@ export default class BanFetcher {
             expires: expires,
             expired: !(ban.attributes.expires === null || expires.getTime() > Date.now()),
 
-            reason: classifyBanReason(`${ban.attributes.reason} ${ban.attributes.note}`),
-            rawReason: `${ban.attributes.reason} ${ban.attributes.note}`,
+            reason: classifyBanReason(ban.attributes.reason,ban.attributes.note),
+            rawReason: ban.attributes.reason,
+            rawNotes: ban.attributes.note,
```
Comment (ect0s) [inline ban-importer/src/ban-fetcher.js:127]: rawNotes here is not part of the DB schema

The object we are building here eventually bubbles down to what we store in the DB.

This leaves 3 options:

Update the Schema and DB.
Omit the Notes.
Store the rawReason as it was previously in the db (ban.reason + ban.notes)


When testing classifier changes, I rescan the raw reason fields as stored in the DB, with this change we will lose that ability unless we update the schema. 

https://github.com/CommunityBanList/Communitybanlist/blob/master/scbl-lib/db/models/ban.js#L10

File: scripts/update-unknown.js [hunk 5]
```diff
@@ -8,11 +8,12 @@ async function main() {
   console.log(`${bans.length} Unknown Bans To Classify`);
   for (const ban of bans) {
     const old = ban.reason;
-    ban.reason = classifyBanReason(ban.rawReason);
+    ban.reason = classifyBanReason(ban.rawReason, ban.rawNote);
```
Comment (ect0s) [inline scripts/update-unknown.js:11]: As above, ban.rawNote would need to be added to the DB schema.

**Review by ect0s — CHANGES_REQUESTED** [review]

File: ban-importer/src/ban-fetcher.js (hunk 4 above)
Comment (Michael-Phoenix) [reply ban-importer/src/ban-fetcher.js:127]: You are correct in that. I think we could get away with it without a DB schema change because we only reclassify unknown bans anyway, am I right?
In that case we could just put notes and reason into the rawReason field in the DB like it was before. Danger here being if we change that aspect further down the line, we might run into unforeseen consequences with this.

On the other hand, it would be more clean to separate them imo.
How hard is it to update the schema?

File: scripts/update-unknown.js (hunk 5 above)
Comment (Michael-Phoenix) [reply scripts/update-unknown.js:11]: Added to DB Schema in https://github.com/CommunityBanList/Communitybanlist/pull/11/commits/031fe5c50d15c57831a1b3df725a2d08fa9a098f

**Review by ect0s — APPROVED** [review]
